File: LOO 2°Parcial/Tarea 2/Biblioteca Personal/Biblioteca Personal/Caracteristicas.cpp
#include "Caracteristicas.h"
Caracteristicas::Caracteristicas() {
	this->title = "";
	this->ci = 0;
	this->year = 0;
}

Caracteristicas::Caracteristicas(string title, int ci, int year) {
	this->setTit(title);
	this->setCod(ci);
	this->setAnio(year);
}
void Caracteristicas::setTit(string title) {
	this->title = title;
}
string Caracteristicas::getTit(void) { return this->title; }

void Caracteristicas::setCod(int ci) {
	this->ci = ci < 0 ? 0 : ci;
}
int Caracteristicas::getCod(void) { return this->ci; }

void Caracteristicas::setAnio(int year) {
	this->year = year > 0 && year < 9999 ? year : 1;
}
int Caracteristicas::getAnio(void) { return this->year; }

void Caracteristicas::imprimir(void) {
	cout << "\nTitulo: " << this->title << "\nCodigo de identificacion: " << this->ci << "\nAnio de publicacion: " << this->year <<"\n";
}

File: LOO 2°Parcial/Tarea 2/BibliotecaJava/src/bibliotecajava/Caracteristicas.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package bibliotecajava;

/**
 * 
 * @author Dimitri Medina Novelo
 */
public class Caracteristicas {
    protected String title; 
    protected int ci, year;
    
    public Caracteristicas() {
        this.title="";
        this.ci=0;
        this.year=0;
    }
    public Caracteristicas(String title, int ci, int year) {
        this.setTit(title);
        this.ci=(ci);
        this.year=(year);
    }
    public String getTit() { return this.title; }
    public void setTit(String value) { this.title = value; }
    
    public int getCOD() {return this.ci;}
    public void setCOD(int value) { this.ci = ci < 0 ? 0 : ci;}
    
    public int getYear() {return this.year;}
    public void setYear(int value) { this.year = year > 0 && year < 9999 ? year : 1;}
    //"\n"
    
    public void imprimir() {
        System.out.print("Titulo: " + this.title + "\nCodigo de identificacion: " + this.ci + "\nAnio de publicacion: " + this.year+"\n");
    }
}

